### ISSUE TITLE
Document Viewer is not reloading

### ISSUE THREAD

**Issue by acronymm** [trailheadapps/apex-recipes#384]
### Summary

When opened first time, Documentation tab on Apex Recipes page works properly. It loads the related .md file from resources. But it doesn't reload if you select another class. It keeps showing the document of the class that was previously selected.

### Salesforce Org Type

Trailhead Playground

### Steps To Reproduce

1. Get a new Trailhead Playground.
2. Deploy this project to the org following installation guide provided in README.
3. Open the org and go to Apex Recipes app.
4. From tree-view select any class. (e.g., AtFutureRecipes)
5. Click the Documentation tab. It works as expected.
![Screenshot 2023-01-13 at 12 28 37 PM](https://user-images.githubusercontent.com/44998728/212230572-f6bdebc4-2fc2-4d9c-a229-2fed44196694.png)
6. Now select another class from the tree-view. (e.g., BatchApexRecipes)
7. Documentation contents do not reload and keep showing the previous class.
![Screenshot 2023-01-13 at 12 31 57 PM](https://user-images.githubusercontent.com/44998728/212230907-7ca11551-2bd7-4044-8405-2076a0796d3a.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by pozil** [comment]
HI @acronymm and thanks for reporting this issues and for opening a PR to fix it.
I had to write a slightly different fix that what you did so I created a separate PR #386 and I'll close yours.

In short, the viewer was not reloading the docs because it only loaded a doc page once in the `renderedCallback` lifecycle hook (the hook is only called the first time the component is displayed). It was a design mistake to place the `fetch` in there so I moved it to a new `loadDocumentation` method. This methods is now called by the setter for the `recipeName` property. This mean that each time we change `recipeName` we load the right doc content.
